### PR TITLE
Remove adm-zip dependency

### PR DESCRIPTION
**Pull request by dpgraham**
* Wrote a test for `commands.pullFolder` that verifies it works correctly with adm-zip
* Removed adm-zip dependency and made the test fail
* Added appium support zip dependency and replaced it with that and made the tests work again

### CODE REVIEW

File: test/unit/commands/actions-specs.js [hunk 1]
```diff
@@ -48,4 +50,41 @@ describe('Actions', () => {
       driver.bootstrap.sendAction.calledWith('drag').should.be.true;
     });
   });
+  describe('pullFolder', () => {
+    it('should pull a folder and return base64 zip', async () => {
+
+      let adbPullStub;
+      const pull = async (ignore, localPath) => {
+        await mkdirp(path.resolve(localPath));
```
Comment (imurchie) [inline test/unit/commands/actions-specs.js]: Why not just make this a fixture in `test/assets`?

File: test/unit/commands/actions-specs.js [hunk 2]
```diff
@@ -48,4 +50,41 @@ describe('Actions', () => {
...
+      }
+
+      // Get the zipped buffer contents and write it to a .zip file
+      const buffer = await driver.pullFolder('/does/not/matter');
+      (typeof(buffer)).should.equal('string');
+      const zippedDir = await tempDir.openDir();
+      await fs.writeFile(path.resolve(zippedDir, 'zipped.zip'), buffer, 'base64');
+      
+      // Extract the zip file
+      const unzippedDir = await tempDir.openDir();
```
Comment (imurchie) [inline test/unit/commands/actions-specs.js]: Can't this all be done in streams or something so there doesn't need to be file system writes in a unit test?

Comment (dpgraham) [inline test/unit/commands/actions-specs.js]: I found a library called `mock-fs` that overrides `fs` and does everything in memory. I think I'll use that as a dev-dependency.

File: test/unit/commands/actions-specs.js (hunk 1 above)
Comment (dpgraham) [inline test/unit/commands/actions-specs.js]: This was actually unnecessary. I took it out.

**Comment by dpgraham** [comment]
Changes made. Uses `mock-fs` library as a dev-dependency.

File: package.json [hunk 3]
```diff
@@ -72,7 +71,9 @@
     "eslint-plugin-babel": "^3.3.0",
     "eslint-plugin-import": "^1.9.2",
     "eslint-plugin-mocha": "^3.0.0",
+    "fancy-log": "^1.3.0",
```
Comment (dpgraham) [inline package.json]: I got rid of it. There was an error that happened earlier that said that the module was missing so I tried fixing it by adding it as a dev-dep. That was unneccessarry. I just did `rm -rf node_modules && npm install && gulp once` and it worked out fine.

Comment (jlipps) [inline package.json]: 👍

**Comment by dpgraham** [comment]
Tagged and published to 1.17.4 (https://www.npmjs.com/package/appium-android-driver)

**Pull request merged**